docs/developer.rst

```diff
@@ -0,0 +1,120 @@
+
+   .. code-block:: jinja
+
+      {% include 'my_other_file.html' %}
+
+performance-wise this is not always a good idea. Each and every time the 
+'include' statement is read while the page is constructed the included 
+file has to be read from the filesystem. File reads are (relatively) slow and 
+if the include is inside of a loop with a large number of iterations you can 
+easily add extra seconds of load time for a minimal boost in HTML readability.
+
+Some tips to get the most out of Jinja without adding too much overhead:
+  - If you have a loop and want to 'include' the body of the loop from another
+    file, it's better to keep the loop inside the included file (so the file is
+    opened and read once, rather than once per iteration)
+  - If you have an 'if' statement and the body of it is included from another
+    snippet, it's better to keep the 'if' in your original file, so you don't
+    need to open the snippet just to discover the if statement failed
+
+Also note that if you're organizing your html snippets in a nested folder you 
+always need to give the full path from the root of the template directory to 
+the file you want to include. If you get an error about a missing template, 
+make sure you quoted the name of the file to be included.
+
+  .. code-block:: jinja
+
+    # Good
+    {% include 'my_snippet.html' %} # for templates/my_snippet.html
+    {% include 'modals/incidental_findings.html' %} # For templates/modals/incidental_findings.html
+
+    # Bad
+    {% include my_snippet.html %} # Missing quotes on file name
+    {% include 'incidental_findings.html' %} # This file won't be found without the full path
+   
+
+SQLAlchemy Tips
+---------------
+SQLAlchemy is awesome and very powerful BUT sometimes it makes really naive 
+queries. If you try to work with objects from dashboard.models like they're 
+normal python objects you can very easily end up generating thousands of queries 
+without realizing. For instance if you tried doing something like this:
+
+  .. code-block:: python
+
+    from dashboard.models import Site
+
+    cmh = Site.query.get('CMH')
+
+    for timepoint in cmh.timepoints:
+       # Do some stuff with the timepoint record here
+
+This loop would generate one query to the database for each timepoint that has
+the site 'CMH'. If code like that were embedded in a function called in a jinja
+template, then those extra queries would hit every time a user loaded the page.
+In these sorts of cases it is often better to craft your own queries using
+`SQLAlchemy's query API. <http://docs.sqlalchemy.org/en/latest/orm/query.html>`_
+
+
+Debugging Tips
+--------------
+* To help with debugging consider using the `Flask Debug Toolbar. <https://flask-debugtoolbar.readthedocs.io/en/latest/index.html>`_
+  It can help identify what's generating most of the load time and help you 
+  identify if your SQLAlchemy code is accidentally generating a large number 
+  of queries.
```

Comment:
```suggestion
  It can assist in identifying bottlenecks in load time caused by excess querying by giving you a breakdown of your SQLAlchemy queries.
```